src/buttons/src/IconButton.js

```diff
@@ -2,14 +2,32 @@ import React, { memo, forwardRef } from 'react'
 import PropTypes from 'prop-types'
 import { dimensions, spacing, position, layout } from 'ui-box'
 import { IconWrapper } from '../../icons/src/IconWrapper'
-import useButtonAppearance from '../../theme/src/hooks/useButtonAppearance'
-import Button, { getIconSizeForButton } from './Button'
+import useStyleConfig from '../../hooks/use-style-config'
+import Button, {
+  getIconSizeForButton,
+  internalStyles,
+  pseudoSelectors
+} from './Button'
 
 const IconButton = memo(
   forwardRef(function IconButton(props, ref) {
-    const { icon, iconSize, size = 'medium', ...restProps } = props
-    // TODO figure out a better solution for dynamically relative values
-    const { boxProps } = useButtonAppearance({ size })
+    const {
+      icon,
+      iconSize,
+      size = 'medium',
+      intent,
+      appearance,
+      ...restProps
+    } = props
+
+    // Composes the exact same styles as button
+    const { className: themedClassName, ...boxProps } = useStyleConfig(
+      'Button',
+      { appearance, intent, size },
```

Comment:
Thinking for consumers we _may_ want to pass the entire props object... but tbd. Kind of false advertising if we dont

Reply:
Yeah, probably - would make writing themes be the most complete. I guess we can revisit and see if we want to or not!